### PR TITLE
Polkadot v0.9.36

### PR DESCRIPTION
**Pull request by cdamian**
This branch was created from the one used in https://github.com/centrifuge/fudge/pull/55.

### CODE REVIEW

File: core/src/provider/initiator.rs [hunk 1]
```diff
@@ -200,13 +198,6 @@ where
 		self
 	}
 
-	/// Overwrites the used keystore pointer that will be used when initiating the
-	/// structs for a core builder.
-	pub fn with_keystore(&mut self, keystore: SyncCryptoStorePtr) -> &mut Self {
-		self.keystore = Some(keystore);
-		self
-	}
```
Comment (mustermeiszer) [inline core/src/provider/initiator.rs:208]: This must not be removed. The keystore is now inner part of the `ExeutionExtension` struct

File: core/src/provider/initiator.rs [hunk 2]
```diff
@@ -168,21 +164,23 @@ where
...
-			execution_strategies: ExecutionStrategies::default(),
+			execution_extensions: ExecutionExtensions::default(),
 		}
 	}
 
-	/// Overwrites the used `ExecutionStrategies` that will be used when initiating the
+	/// Overwrites the used `ExecutionExtensions` that will be used when initiating the
 	/// structs for a core builder.
-	pub fn with_exec_strategies(&mut self, execution_strategies: ExecutionStrategies) -> &mut Self {
-		self.execution_strategies = execution_strategies;
```
Comment (mustermeiszer) [inline core/src/provider/initiator.rs:185]: I would also not remove that but rather have both of the methods availabvle here.

Exeutions strategies is also part of `ExecutionExtension`

**Review by mustermeiszer — CHANGES_REQUESTED** [review]
See comments. Otherwise good!

File: core/src/provider/initiator.rs (hunk 1 above)
Comment (cdamian) [reply core/src/provider/initiator.rs:208]: Yes, but that's why we now have `with_exec_extensions`. One can provide the keystore and/or strategies this way.

Alternatively, we could also add:

```
	pub fn with_keystore(&mut self, keystore: SyncCryptoStorePtr) -> &mut Self {
		let ext = ExecutionExtensions::new(ExecutionStrategies::default(), Some(keystore), None);
		self.with_exec_extensions(ext)
	}

	pub fn with_exec_strategies(&mut self, strategies: ExecutionStrategies) -> &mut Self {
		let ext = ExecutionExtensions::new(strategies, None, None);
		self.with_exec_extensions(ext)
	}
```

WDYT?

**Pull request merged**